test/e2e/e2e_ginkgo/utilg_test.go

```diff
@@ -0,0 +1,76 @@
+package e2e_ginkgo
+
+import (
+	"fmt"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/apiserver/pkg/storage/names"
+	"time"
+
+	"github.com/operator-framework/operator-lifecycle-manager/pkg/api/client"
+	"github.com/operator-framework/operator-lifecycle-manager/pkg/api/client/clientset/versioned"
+	"github.com/operator-framework/operator-lifecycle-manager/pkg/lib/operatorclient"
+	"github.com/sirupsen/logrus"
+)
+
+var (
+	cleaner                    *namespaceCleaner
+	genName                          = names.SimpleNameGenerator.GenerateName
+	immediateDeleteGracePeriod int64 = 0
+)
+
+const (
+	pollInterval = 1 * time.Second
+	pollDuration = 5 * time.Minute
+)
+
+type namespaceCleaner struct {
+	namespace      string
+	skipCleanupOLM bool
+}
+
+type checkResourceFunc func() error
+
+func newNamespaceCleaner(namespace string) *namespaceCleaner {
+	return &namespaceCleaner{
+		namespace:      namespace,
+		skipCleanupOLM: false,
+	}
+}
+
+func newKubeClient() operatorclient.ClientInterface {
+	if kubeConfigPath == nil {
+		fmt.Printf("using in-cluster config")
+	}
+	// TODO: impersonate ALM serviceaccount
+	// TODO: thread logger from test
+	return operatorclient.NewClientFromConfig(*kubeConfigPath, logrus.New())
+}
+
+func newCRClient() versioned.Interface {
```

Comment:
again, all of these functions won't be exportable. consider changing file name from _test.go to _util.go or something. see https://medium.com/@robiplus/golang-trick-export-for-test-aa16cbd7b8cd

Reply:
done